ethereum/eth/src/test/java/tech/pegasys/pantheon/ethereum/eth/manager/EthProtocolManagerTest.java

```diff
@@ -744,4 +755,38 @@ public void shouldSuccessfullyRespondToGetHeadersRequestLessThanZero()
       done.get();
     }
   }
+
+  @Test
+  public void transactionMessagesGoToTheCorrectExecutor() {
```

Comment:
nitpick - this might make more sense in the `TransactionPoolFactory` test, as the tx pool is the object deciding how to use the workers

Reply:
The message dispatch occurs within the EthProtocolManager, even though TransactionPool is the one registering the listener.  EthProtocolManager::processMessage calls are key to this test.  Also, this would result in core then depending on eth (at least the tests) and would result in a quasi-circular dependency.  While it does bring the test closer to the other half of the code that is being tested EthScheduler is also as much under test.  But the number of new quasi-circular imports that :ethereum:core_test would gain if I moved the test persuades me to leave it where it is.